tests/acceptance/test_config_refresh.py

```diff
@@ -67,3 +74,61 @@ def test_config_refresh_after_5_seconds_without_change(tmp_path, config):
             "Successfully reloaded configuration",
             test_timeout=10,
         )
+
+
+def test_config_refresh_after_crash_config_not_changed(tmp_path, config):
+    config.config_refresh_interval = 5
+    config.metrics = {"enabled": False}
+
+    config_path = tmp_path / "generated_config.yml"
+    config_path.write_text(config.as_json())
```

Comment:
Please add the original value of the pipeline here (or put it with the lines 80/81). It's way clearer if the pre and post state are both visible inside (or close to) the test method using it.

```suggestion
    config_path = tmp_path / "generated_config.yml"
    config.pipeline = # original value here
    config_path.write_text(config.as_json())
```